src/ir/effects.h

```diff
@@ -61,10 +61,11 @@ struct EffectAnalyzer
   // of control flow proceeding normally).
   bool branchesOut = false;
   bool calls = false;
-  std::set<Index> localsRead;
-  std::set<Index> localsWritten;
-  std::set<Name> globalsRead;
-  std::set<Name> globalsWritten;
+  std::unordered_set<Index> localsRead;
+  std::unordered_set<Index> localsWritten;
+  std::unordered_set<Name> globalsRead;
+  std::unordered_set<Name> globalsWritten;
+  std::unordered_set<Name> breakTargets;
```

Comment:
Did you measure this? I worry it might hurt actually. I suspect that mostly there are very few locals or globals, and so a set may be faster than an unordered one (the unordered one computes a hash).

Reply:
you're right, Benckmark with `-O4` for `1.1 mb` file:

master:

1.950 total
1.929 total
1.894 total
1.878 total 1.859 total
1.897 total

PR without unordered_set but with Const/Nop checks:  1.898 total
1.951 total
1.932 total
1.935 total
1.930 total
1.969 total
1.972 total

PR with unordered_set but without Const/Nop checks:  2.061 total
2.115 total
2.210 total
2.063 total
2.170 total

So this improvements basically doesn't improve anything. Will close this)